### PR TITLE
Merchant Order Item Fulfillment

### LINKED ISSUE #30: User Story 53, Merchant fulfills part of an order
```
As a merchant
When I visit an order show page from my dashboard
For each item of mine in the order
If the user's desired quantity is equal to or less than my current inventory quantity for that item
And I have not already "fulfilled" that item:
- Then I see a button or link to "fulfill" that item
- When I click on that link or button I am returned to the order show page
- I see the item is now fulfilled
- I also see a flash message indicating that I have fulfilled that item
- My inventory quantity is permanently reduced by the user's desired quantity

If I have already fulfilled this item, I see text indicating such.
```

DO THESE STORIES FIRST:
- User Story 52, Merchant sees an order show page

### Checklist:

- [x] base tests are written
- [x] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #212

child of #274

depends on #211

### PR DESCRIPTION
**Pull request by BrennanAyers**
This PR brings in the functionality for a Merchant to Fulfill Items on Orders they have Items in. This is done through the Dashboard Order Show Page, where a button will be enabled if they have the ability to Fulfill the Item.
For the button to be enabled, the Merchant has to have not already fulfilled the Item and have enough of the Item in their Item Inventory. If they do not have enough Inventory to Fulfill the Item, they will be shown a message stating as much, and the button will be disabled. Similarly, if the Item has already been Fulfilled, they will be shown a message letting them know and the button will be disabled.
This functionality brought to light an issue with the OrderItems model that was using `after_save` callback functions. These functions were being called whenever an OrderItem was saved, including when it was created. This required the `unless:` clause, to check if the `created_at` and `updated_at` values were the same, indicating a Create action, as opposed to an Update action on the resource. Tests have been updated to expect the correct values.

Resolves #30 #29

### CODE REVIEW

File: app/models/order_item.rb [hunk 1]
```diff
@@ -33,4 +33,8 @@ def check_order
       end
     end
   end
+
+  def just_created?
+    created_at == updated_at
+  end
 end
```
Comment (VinceCarollo) [inline app/models/order_item.rb:40]: Smart fix

**Review by VinceCarollo — APPROVED** [review]
